impl/src/main/java/org/jboss/weld/bean/proxy/GroovyMethodFilter.java

```diff
@@ -54,15 +55,29 @@ public boolean isEnabled() {
     }
 
     @Override
-    public boolean accept(Method method) {
+    public boolean accept(Method method, Class<?> proxySuperclass) {
         if (GROOVY_OBJECT.equals(method.getDeclaringClass().getName())) {
             return false;
         }
-        for (MethodSignature groovyMethod : METHODS) {
-            if (groovyMethod.matches(method)) {
-                return false;
+        if (isGroovyObject(proxySuperclass)) {
```

Comment:
What's the difference between `isGroovyObject(proxySuperclass)` and `GROOVY_OBJECT.equals(method.getDeclaringClass().getName())` (three lines before)?